### PR TITLE
Add azure-xml dependency to azure-core

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Adds `azure-xml` as a dependency to `azure-core` and updates internal logic to use the APIs is `azure-xml` directly instead of with reflection.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core-serializer-json-jackson/pom.xml [hunk 1]
```diff
@@ -81,6 +81,11 @@
       <artifactId>azure-json</artifactId>
       <version>1.1.0</version> <!-- {x-version-update;com.azure:azure-json;dependency} -->
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-xml</artifactId>
+      <version>1.0.0-beta.4</version> <!-- {x-version-update;unreleased_com.azure:azure-xml;dependency} -->
+    </dependency>
```
Comment (srnagar) [inline sdk/core/azure-core-serializer-json-jackson/pom.xml:88]: Why do we need the xml dependency in `json-jackson` library?

File: sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonAdapter.java [hunk 2]
```diff
@@ -123,13 +129,13 @@ public byte[] serializeToBytes(Object object, SerializerEncoding encoding) throw
 
         if (encoding == SerializerEncoding.XML) {
             return supportsXmlSerializable(object.getClass())
-                ? ReflectionSerializable.serializeXmlSerializableToBytes(object)
+                ? serializeXmlSerializableToBytes((XmlSerializable<?>) object)
```
Comment (srnagar) [inline sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonAdapter.java:132]: Ah - we have both xml and json support in json-jackson package. Package name is misleading.

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java [hunk 3]
```diff
@@ -241,7 +179,34 @@ public static Object deserializeAsJsonSerializable(Class<?> jsonSerializable, by
...
      */
     public static boolean supportsXmlSerializable(Class<?> bodyContentClass) {
-        return XML_SERIALIZABLE_SUPPORTED && XML_SERIALIZABLE.isAssignableFrom(bodyContentClass);
+        if (FROM_XML_CACHE.containsKey(bodyContentClass)) {
+            return true;
+        }
+
+        if (!XmlSerializable.class.isAssignableFrom(bodyContentClass)) {
+            return false;
+        }
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java:188]: Isn't it sufficient to check whether the class is implementing XmlSerializable? Do we need to check if fromXml and toXml methods are present?

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java [hunk 4]
```diff
@@ -32,78 +33,11 @@
...
-        XML_SERIALIZABLE = xmlSerializable;
-        XML_READER = xmlReader;
-        XML_READER_CREATOR = xmlReaderCreator;
-        XML_WRITER_CREATOR = xmlWriterCreator;
-        XML_WRITER_WRITE_XML_START_DOCUMENT = xmlWriterWriteStartDocument;
-        XML_WRITER_WRITE_XML_SERIALIZABLE = xmlWriterWriteXmlSerializable;
-        XML_WRITER_FLUSH = xmlWriterFlush;
-        XML_SERIALIZABLE_SUPPORTED = xmlSerializableSupported;
-        FROM_XML_CACHE = XML_SERIALIZABLE_SUPPORTED ? new ConcurrentHashMap<>() : null;
+        FROM_XML_CACHE = new ConcurrentHashMap<>();
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java:40]: Should we have a limit on the size of this cache?

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java (hunk 3 above)
Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java:188]: This is done to handle cases where a class extends a class implementing XmlSerializable but it doesn't implement the interface methods.

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java (hunk 4 above)
Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java:40]: We do have a limit: https://github.com/Azure/azure-sdk-for-java/pull/39524/files/b31df7782f3afb2650483028ffe689b5a2f4771e#diff-81c191c5c4f7986f23f1c2a022f92d0babdbca2bdd3b2f623493fbccccc6dd74R289

**Review by srnagar — APPROVED** [review]

**Pull request merged**